packages/grid/x-data-grid/src/tests/selection.DataGrid.test.tsx

```diff
@@ -411,6 +411,19 @@ describe('<DataGrid /> - Selection', () => {
       );
       expect(getSelectedRowIds()).to.deep.equal([]);
     });
+
+    it('should not crash when paginationMode="server" and some selected rows are not provided to the grid', () => {
+      render(
```

Comment:
`.not.to.throw()`?

https://github.com/mui-org/material-ui-x/blob/f581ac12a3dfc9097f3877b3334a935b5fdbfdac/packages/grid/x-data-grid-pro/src/tests/rows.DataGridPro.test.tsx#L773

Reply:
On `master` this code throw
I guess we can just render and it would fail if it throw anyway.
My goal was to be extra-clear here.

Reply:
I was referring to the error message. By not using `.not.to.throw()`, the test fails because an error was written to the console, not because the test itself failed.

Reply:
Isn't the console error catch by our custom `toErrorDev` instead of `to.throw` ?

Reply:
Yeah, true. Either `.toErrorDev` or `to.throw` works. My point is more that this test has no assertion and relies on the default behavior of failing if `console.error` is called.

Reply:
Yes indeed
If you have a better idea for this kind of test where we just want to avoid a regression with a crash.

Reply:
Use `.toErrorDev()`? There're other examples in the "console error" describe block in this same file.

```tsx
expect(() => {
  render(
    <TestDataGridSelection
      paginationMode="server"
      selectionModel={[1, 4]}
      isRowSelectable={(params) => params.id > 0}
      checkboxSelection
    />,
  );
  // @ts-expect-error need to migrate helpers to TypeScript
}).not.toErrorDev();
```

Reply:
```
      expect(() => {
        render(<TestCase initialState={{ pinnedColumns: { left: ['currency'] } }} />);
        render(<TestCase initialState={{ pinnedColumns: { right: ['currency'] } }} />);
      }).not.to.throw();
```

For column pinning you used `not.to.throw`
But OK to prefer `not.toErrorDev`

Reply:
And I understand what you were saying, I thought I did `not.to.throw` but I did neither of them so the test was confusion.
I just added `not.toErrorDev`